404 request when comparing two products

**To Reproduce**
- Go to the _Products_ report.
- Select _Product Comparison_ as the filter.
- Select two products to compare.
- Click on `Compare`.
- Check the `Network` tab in your browser devtools and verify there is a call to the endpoint `/wc/v4products` (notice the missing slash) that returned 404.
![image](https://user-images.githubusercontent.com/3616980/51979257-f7078f80-248c-11e9-9eb5-cbfee4c97805.png)

I can still reproduce the issue both in `master` and `add/product-detail-report`:

![peek 2019-01-31 14-41](https://user-images.githubusercontent.com/3616980/52057802-52598080-2566-11e9-91ef-d12e4b8f3fcb.gif)

I needed to merge `master` into `add/product-detail-report` to get it working, otherwise it was not building with this error:
```
[0] ERROR in ./packages/currency/build-module/index.js
[0] Module not found: Error: Can't resolve '@woocommerce/number' in '/var/www/html/woocommerce/wp-content/plugins/wc-admin/packages/currency/build-module'
```

I don't think that's related at all, but just wanted to point it out.

---

#1425 doesn't seem to be fixed for me either in any of both branches.

Am I missing something?  :confused:

--- a/client/lib/async-requests/index.js
+++ b/client/lib/async-requests/index.js
@@ -39,30 +39,30 @@ export function getRequestByIdString( path, handleData = identity ) {
 }
 
 export const getCategoryLabels = getRequestByIdString(
-	NAMESPACE + 'products/categories',
+	NAMESPACE + '/products/categories',
 	category => ( {
 		id: category.id,
 		label: category.name,
 	} )
 );
 
-export const getCouponLabels = getRequestByIdString( NAMESPACE + 'coupons', coupon => ( {
+export const getCouponLabels = getRequestByIdString( NAMESPACE + '/coupons', coupon => ( {
 	id: coupon.id,
 	label: coupon.code,
 } ) );
 
-export const getCustomerLabels = getRequestByIdString( NAMESPACE + 'customers', customer => ( {
+export const getCustomerLabels = getRequestByIdString( NAMESPACE + '/customers', customer => ( {
 	id: customer.id,
 	label: customer.username,
 } ) );
 
-export const getProductLabels = getRequestByIdString( NAMESPACE + 'products', product => ( {
+export const getProductLabels = getRequestByIdString( NAMESPACE + '/products', product => ( {
 	id: product.id,
 	label: product.name,
 } ) );
 
 export const getVariationLabels = getRequestByIdString(
-	query => NAMESPACE + `products/${ query.products }/variations`,
+	query => NAMESPACE + `/products/${ query.products }/variations`,
 	variation => {
 		return {
 			id: variation.id,

